basic-services/README.md

```diff
@@ -0,0 +1,176 @@
+Console.Log(result.NumericValue)
+
+// Read an enum value
+var result = client.ReadProperty(guid2, "presentValue")
+var enumValue = result.StringValue
+Console.Log(resourceManger.GetString(enumValue, "en-US")
+
+// Read a string value
+var result = client.ReadProperty(guid3, "description")
+Console.Log(result.StringValue)
+
+// Read a boolean value
+var result = client.ReadProperty(guid4, "isEnabled")
+Console.Log(result.StringValue) // prints "True" or "False" 
+Console.Log(result.NumericValue) // prints 1 or 0
+Console.Log(result.BooleanValue) // prints "True" or "False". This BooleanValue isn't in scope but perhaps it should be
+
+// Read an array value, new for basic services, didn't exist in MSSDA
+var result = client.ReadProperty(guid5, "ipAddress") // ipAddress is an array of bytes
+var firstByteResult = result.ArrayValue
+Console.Log(firstByteResult.Item1)   // prints "192" as Item1 is the string representation
+
+// alternatively
+var (string StringValue, double NumericValue) result = client.ReadProperty(guid5, "ipAddress") // ipAddress is an array of bytes
+Console.Log(result.StringValue)
+
+```
+
+
+Requirements
+-----------
+
+This project will have several small phases. The first phase will
+expose add reading of objects/attributes. The second phase will writing and commanding.
+The final phase will add discover of devices and objects.
+
+* The project must maintain a CHANGELOG (https://keepachangelog.com/en/1.0.0/).I have example here: https://github.com/metasys-server/nodekit/blob/master/CHANGELOG.md 
+* This package must be published in the metasys organization at nuget.org with appropriate semantic versioning, https://semver.org
+copyright, licensing, link to github repo, etc. 
+
+* We will release a version at the end of Phase 1. Most likely 1.0. Earlier versions can be published but only as 0.x versions
+or as pre-release versions of 1.0 (follow microsoft guidance here on prerelease versioning: https://docs.microsoft.com/en-us/nuget/create-packages/prerelease-packages )
+    Breaking changes should align with major versions of Metasys where possible. We should avoid any breaking changes
+    in between release of Metasys.
+    
+* Naming guidelines: Follow Microsoft Naming guidelines https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines 
+
+* All submitted code must be reviewed through pull requests.
+
+* Good commit messages must be used on all commits. https://chris.beams.io/posts/git-commit/ 
+Most commits should NOT be one liners. When adding a commit add a paragraph or a few paragraphs explaining what you did.
+Only the smallest of commits should be one line messages.
+
+* Consider adding a light weight CONTRIBUTING.md or referencing Microsoft's for guidance for outsiders and others in JCI.
+
+* Track the Issues on this repo and provide guidance (where possible) on roadmap.
+
+### Phase 1
+
+We will begin on what I'm calling the TraditionalClient. (name can be changed).
```

Comment:
Will the scope of this project include having the traditional (basic) client and a more advanced client that has more features? I can see having two different clients being beneficial in this case.

Reply:
Yeah, that was my original reason for using this naming. The other would be the full featured one, using async, using generics, exposing more options, etc.

Reply:
Although, if we have both clients in this library, then the library shouldn't be called `basic-services`.

I was thinking about calling it something like `metasys-services` or `metasys-client` or just some codename.

I suppose `metasys-client-dotnet` is the best name for this. Then it can have TraditionalClient (or BasicClient) and then just `MetasysClient` types.